searx/network/__init__.py

```diff
@@ -1,266 +1,137 @@
 # SPDX-License-Identifier: AGPL-3.0-or-later
 # lint: pylint
-# pylint: disable=missing-module-docstring, global-statement
-
-import asyncio
+# pylint: disable=global-statement
+"""Provide HTTP clients following the user configuration
+"""
 import threading
-import concurrent.futures
-from queue import SimpleQueue
-from types import MethodType
-from timeit import default_timer
-from typing import Iterable, NamedTuple, Tuple, List, Dict, Union
 from contextlib import contextmanager
+from typing import Callable, Optional
 
 import httpx
-import anyio
 
-from .network import get_network, initialize, check_network_configuration  # pylint:disable=cyclic-import
-from .client import get_loop
-from .raise_for_httperror import raise_for_httperror
+from searx.network.context import NetworkContext, P, R
+from searx.network.network import NETWORKS
+from searx.network.raise_for_httperror import raise_for_httperror
+
+__all__ = [
+    "NETWORKS",
+    "get_http_runtime",
+    "set_networkcontext_for_thread",
+    "clear_networkcontext_for_thread",
+    "networkcontext_for_thread",
+    "networkcontext_call",
+    "request",
+    "raise_for_httperror",
+    "get",
+    "options",
+    "head",
+    "post",
+    "put",
+    "patch",
+    "delete",
+]
+
+
+_THREADLOCAL = threading.local()
+"""Thread-local that contains only one field: network_context."""
+
+_NETWORK_CONTEXT_KEY = 'network_context'
+"""Key to access _THREADLOCAL"""
+
+
+class NetworkContextNotFound(Exception):
+    """A NetworkContext is expected to be set in this thread.
+
+    Use searx.network.set_context_for_thread or searx.network.context_for_thread
+    to set a NetworkContext
+    """
 
 
-THREADLOCAL = threading.local()
-"""Thread-local data is data for thread specific values."""
+def get_http_runtime() -> Optional[int]:
```

Comment:
Rename to `get_http_runtime_for_thread` to keep coherency with other method names.

 Move this method after `networkcontext_call`